pytest_bdd/utils.py

```diff
@@ -84,3 +84,29 @@ def get_request_fixture_names(request):
     Compatibility with pytest 3.0.
     """
     return request._pyfuncitem._fixtureinfo.names_closure
+
+
+def get_parametrize_markers_args(node):
+    """In pytest 3.6 new API to access markers has been introduced and it deprecated
+    MarkInfo objects.
+
+    This function uses that API if it is available otherwise it uses MarkInfo objects.
+    """
+    mark_name = 'parametrize'
+    try:
+        return get_markers_args_using_iter_markers(node, mark_name)
+    except AttributeError:
+        return get_markers_args_using_get_marker(node, mark_name)
+
+
+def get_markers_args_using_iter_markers(node, mark_name):
+    """Recommended on pytest>=3.6"""
+    args = []
+    for mark in node.iter_markers(mark_name):
+        args += mark.args
+    return tuple(args)
+
+
+def get_markers_args_using_get_marker(node, mark_name):
+    """Deprecated on pytest>=3.6"""
+    return getattr(node.get_marker(mark_name), 'args', ())
```

Comment:
why do we use getattr here? `node.get_marker(mark_name).args` should be fine, right?

Reply:
`node.get_marker(mark_name)` may return `None` which does not have `args` so with above approach we may end up with `AttributeError`. That is why I used `getattr`. We could switch to:
```python
marker = node.get_marker(mark_name)
if marker:
    return marker.args
return ()
```
or something similar

Reply:
right, didn't see that. No need to change.